indico/modules/events/contributions/controllers.py

```diff
@@ -322,3 +326,50 @@ def _process(self):
             update_contribution(self.contrib, {'duration': form.duration.data})
             return jsonify_data(new_value=format_human_timedelta(self.contrib.duration))
         return jsonify_form(form, back_button=False, disabled_until_change=True)
+
+
+class RHContributionsMaterialPackage(RHManageContributionsActionsBase, AttachmentPackageGeneratorMixin):
+    """Generate a ZIP file with materials for a given list of contributions"""
+
+    def _process(self):
+        attachments = self._filter_by_contributions({c.id for c in self.contribs}, None)
+        if not attachments:
+            flash(_('The selected contributions do not have any materials.'), 'warning')
+            return redirect(url_for('.manage_contributions', self.event_new))
+        return self._generate_zip_file(attachments)
+
+
+class RHContributionsExportCSV(RHManageContributionsActionsBase):
+    """Export list of contributions to CSV"""
+
+    def _process(self):
+        headers, rows = generate_spreadsheet_from_contributions(self.contribs)
+        return send_csv('contributions.csv', headers, rows)
+
+
+class RHContributionsExportExcel(RHManageContributionsActionsBase):
+    """Export list of contributions to XLSX"""
+
+    def _process(self):
+        headers, rows = generate_spreadsheet_from_contributions(self.contribs)
+        return send_xlsx('contributions.xlsx', headers, rows)
+
+
+class RHContributionsExportPDF(RHManageContributionsActionsBase):
+    def _process(self):
+        pdf = ContribsToPDF(self._conf, self.contribs)
+        return send_file('contributions.pdf', pdf.generate(), 'application/pdf')
+
+
+class RHContributionsExportPDFBook(RHManageContributionsActionsBase):
+    def _process(self):
+        filename = _('{0} - Book of abstracts.pdf').format(self.event_new.title)
```

Comment:
I'd just use `book_of_abstracts.pdf`. The event title makes the filename rather awful and you don't even `secure_filename` it so it'd keep all kinds of garbage not suitable for a filename. In any case, we don't put the event title in filenames or translate them in other places so I wouldn't do it here either.